zmoney/Models/StateManager.swift

```diff
@@ -34,6 +34,7 @@ struct StateManager: StateHolder, StateApplicable {
     var state: State = .noData {
         didSet {
             self.apply(state: state)
+            self.currentState = state
         }
```

Comment:
Немного перемудрил :) 
Почитай про Property observers в Swift https://docs.swift.org/swift-book/LanguageGuide/Properties.html#ID262
В данном случае неплохо подойдет `willSet`, тогда можно сделать так
```suggestion
        willSet {
            self.apply(state: newValue)
        }
```

В таком случае, не нужно свойство `currentState`, так как текущий стейт можно брать из свойства `state`